presto-main/src/main/java/io/prestosql/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -173,6 +188,21 @@ private boolean shouldMergeWithMemory(long memorySize)
     public WorkProcessor<Page> buildResult()
     {
         checkState(hasPreviousSpillCompletedSuccessfully(), "Previous spill hasn't yet finished");
+        producingOutput = true;
+
+        // convert revocable memory to user memory as memory must not be revoked when aggregation operator is finishing
```

Comment:
This isn't really explaining why. Either shorten this to
```
// Convert revocable memory to user memory
```
or add some better explanation. Perhaps (if i got this right):
```
// Convert revocable memory to user memory as returned WorkProcessor holds on to memory so we no longer can revoke.
```

Reply:
The latter is nicer